app/serializers/publication_serializer.rb

```diff
@@ -28,6 +28,36 @@ class PublicationSerializer < PCSSerializer
       end
   end
 
+  attribute :content_blobs do
+    if Seek::Config.allow_publications_fulltext
+      requested_version = object # no version for full text pdf
+
+      if requested_version.respond_to?(:content_blobs)
+        blobs = requested_version.content_blobs
+      elsif requested_version.respond_to?(:content_blob)
+        blobs = [requested_version.content_blob].compact
+      else
+        blobs = []
+      end
+
+      blobs.map { |cb| convert_content_blob_to_json(cb) }
+    end
+  end
+
+  def convert_content_blob_to_json(cb)
```

Comment:
Perhaps this method could be moved from ContributedResourceSerializer to higher up the hierarchy to avoid duplication